src/Mod/Sketcher/App/SketchObject.cpp

```diff
@@ -8335,15 +8375,598 @@ void SketchObject::validateExternalLinks()
     }
 }
 
-void SketchObject::rebuildExternalGeometry(bool defining, bool addIntersection)
+namespace {
+void processEdge2(TopoDS_Edge& projEdge, std::vector<std::unique_ptr<Part::Geometry>>& geos)

```

Comment:
This method needs a more descriptive name.

Reply:
Open to suggestions

Reply:
processEdge2 was initially a part of processEdge (you can see processEdge is calling processEdge2 ). 
I have just split it in half. To make it more readable and also because I needed to call this processEdge2 from somewhere else.

Honestly I don't know what they do exactly and why there are here. I have not modified this, just refactored.
But I agree that it's very opaque and would benefit a good refactoring.
But this is not the goal of this PR.

Reply:
> processEdge2 was initially a part of processEdge (you can see processEdge is calling processEdge2 ). I have just split it in half. To make it more readable and also because I needed to call this processEdge2 from somewhere else.

As such this is very likely. I'd suggest to be careful though that there are no subtle differences between the two calls. Those tests I mentioned elsewhere in this PR would work great.

> Honestly I don't know what they do exactly and why there are here. I have not modified this, just refactored. But I agree that it's very opaque and would benefit a good refactoring. But this is not the goal of this PR.

I'd appreciate if you could dig a little and make a guess. Reduces my headache a little when I have to come around to refactor this later.

Reply:
@AjinkyaDahale Basically it seems that processEdge is general handling of 3d egdes. It is handling some cases manually, then for other types of geos, it's projecting them, then calling processEdge2 on the projected edges.
What is weird is that processEdge2 seems to also project the edge...
Which is why I said I'm unsure what's happening. But what is sure is that trying to simplify breaks everything in my tests. Which is why I left this as untouched as possible.